### PR TITLE
[Driver] LinearOptimisationPriceFinder unit tests

### PR DESCRIPTION
**Pull request by fleupold**
Unit tests for #75

### CODE REVIEW

File: driver/src/price_finding/linear_optimization_price_finder.rs [hunk 1]
```diff
@@ -14,16 +14,25 @@ const RESULT_FOLDER: &str = "./results/tmp/";
...
+    previous_prices: Prices,
+    // default IO methods can be replaced for unit testing
+    write_input: fn(&str, &serde_json::Value) -> std::io::Result<()>,
+    run_solver: fn(&str) -> Result<(), PriceFindingError>,
+    read_output: fn() -> std::io::Result<serde_json::Value>,
 }
 
 impl LinearOptimisationPriceFinder {
     pub fn new() -> Self {
         // All prices are 1 (10**18)
```
Comment (bh2smith) [inline driver/src/price_finding/linear_optimization_price_finder.rs]: Does this mean 1 = 10**18?

**Comment by fleupold** [comment]
🏓 @bh2smith @josojo can you review this please?

File: driver/src/price_finding/linear_optimization_price_finder.rs [hunk 2]
```diff
@@ -168,3 +178,275 @@ fn read_output() -> std::io::Result<serde_json::Value> {
...
+#[cfg(test)]
+pub mod tests {
+
+    use super::*;
+    use std::error::Error;
+
+    #[test]
+    fn test_solver_keeps_prices_from_previous_result() {
+        let state = models::State {
+            state_hash: "hash".to_string(),
```
Comment (bh2smith) [inline driver/src/price_finding/linear_optimization_price_finder.rs]: is this really supposed to be a string with the word hash in it? Or was this a place holder?

File: driver/src/price_finding/linear_optimization_price_finder.rs [hunk 3]
```diff
@@ -43,11 +52,12 @@ fn serialize_order(order: &models::Order, id: &str) -> serde_json::Value {
     })
 }
 
-fn serialize_balances(balances: &Vec<u128>) -> serde_json::Value {
+fn serialize_balances(balances: &Vec<u128>, num_tokens: u8) -> serde_json::Value {
```
Comment (bh2smith) [inline driver/src/price_finding/linear_optimization_price_finder.rs]: Shouldn't this function also have something to do with `num_accounts`?

File: driver/src/price_finding/linear_optimization_price_finder.rs [hunk 4]
```diff
@@ -168,3 +178,275 @@ fn read_output() -> std::io::Result<serde_json::Value> {
...
+            "orders": [
+                {
+                    "execSellAmount": "0",
+                    "execBuyAmount": "0",
+                    "execSurplus": "0"
+                },
+                {
+                    "execSellAmount": "318390084925498118944",
+                    "execBuyAmount": "95042777139162480000",
+                    "execSurplus": "15854632034944469292777429010439194350"
```
Comment (bh2smith) [inline driver/src/price_finding/linear_optimization_price_finder.rs]: Do all of these long numbers have any significance? They all appear to be pretty random. Would is be less distracting to use numbers like `100...00`, `200...00`?

File: driver/src/price_finding/linear_optimization_price_finder.rs (hunk 2 above)
Comment (fleupold) [reply driver/src/price_finding/linear_optimization_price_finder.rs]: It's a unit test, so yeah it's just a dummy string.

File: driver/src/price_finding/linear_optimization_price_finder.rs (hunk 3 above)
Comment (fleupold) [reply driver/src/price_finding/linear_optimization_price_finder.rs]: We don't actually store num_accounts in our system yet. It's assumed that everything we get from the database is a multiple of `TOKENS`, that multiple being num_accounts.

File: driver/src/price_finding/linear_optimization_price_finder.rs (hunk 4 above)
Comment (fleupold) [reply driver/src/price_finding/linear_optimization_price_finder.rs]: Yes, good point.

File: driver/src/price_finding/linear_optimization_price_finder.rs (hunk 3 above)
Comment (bh2smith) [reply driver/src/price_finding/linear_optimization_price_finder.rs]: I believe num_accounts is stored as part of the [SnappInitialization event](https://github.com/gnosis/dex-contracts/blob/4df0570affbe7de86207f790a3a2f8c7f1ccf07b/contracts/SnappBase.sol#L56)

**Comment by bh2smith** [comment]
@josojo 

> Pretty exhausting testing. Great!

This is a quote for the books! I assume you mean "exhaustive", but I agree it was also exhausting!

File: driver/src/price_finding/linear_optimization_price_finder.rs (hunk 3 above)
Comment (josojo) [reply driver/src/price_finding/linear_optimization_price_finder.rs]: Yeah, currently, we are not using these values from the db at all. Also models::Tokens is also just a constant. I would be in favour of making a seperate issue for this and merging this PR

Comment (josojo) [reply driver/src/price_finding/linear_optimization_price_finder.rs]: #86

Comment (bh2smith) [reply driver/src/price_finding/linear_optimization_price_finder.rs]: Cool beans!

**Comment by bh2smith** [comment]
Who Wants to merge!?

**Pull request merged**